File: zxing-android-embedded/src/com/journeyapps/barcodescanner/camera/PreviewCallback.java
package com.journeyapps.barcodescanner.camera;

import com.journeyapps.barcodescanner.SourceData;

/**
 * Callback for camera previews.
 */
public interface PreviewCallback {
    void onPreview(SourceData sourceData);
    void onPreviewError(Exception e);
}

File: zxing-android-embedded/src/com/journeyapps/barcodescanner/camera/PreviewCallback.kt
package com.journeyapps.barcodescanner.camera

import com.journeyapps.barcodescanner.SourceData

/**
 * Callback for camera previews.
 */
interface PreviewCallback {
    fun onPreview(sourceData: SourceData)
    fun onPreviewError(e: Exception?)
}